packages/mermaid/src/diagrams/state/stateDb.js

```diff
@@ -643,10 +639,17 @@ export class StateDB {
   }
 
   getDirection() {
-    return this.direction;
+    const doc = this.rootDoc.find((doc) => doc.stmt === STMT_DIRECTION);
```

Comment:
nit: This can be made a private function `getDirectionStatement`, as it's repeated below.